code/test/tokens.js

```diff
@@ -0,0 +1,472 @@
+      }
+    )
+    // transfer ERC20 tokens from alice.lastResortAddress (which owns the tokens) to alices wallet
+    // await testerc20.transfer(alice.wallet.address, 1000, { from: alice.lastResortAddress })
+    // // alice transfers tokens to bob
+    // await tokenTransfer(
+    //   {
+    //     wallet: alice,
+    //     operationType: ONEConstants.OperationType.TRANSFER_TOKEN,
+    //     tokenType: ONEConstants.TokenType.ERC20,
+    //     contractAddress: testerc20.address,
+    //     dest: bob.wallet.address,
+    //     amount: 100
+    //   }
+    // )
+    // // transfer ERC20d9 tokens from alice.lastResortAddress (which owns the tokens) to alices wallet
+    // await testerc20d9.transfer(alice.wallet.address, 1000, { from: alice.lastResortAddress })
+    // // alice transfers tokens to bob
+    // await tokenTransfer(
+    //   {
+    //     wallet: alice,
+    //     operationType: ONEConstants.OperationType.TRANSFER_TOKEN,
+    //     tokenType: ONEConstants.TokenType.ERC20,
+    //     contractAddress: testerc20d9.address,
+    //     dest: bob.wallet.address,
+    //     amount: 100
+    //   }
+    // )
+    // // transfer ERC721 tokens from alice.lastResortAddress (which owns the tokens) to alices wallet
+    // await testerc721.transferFrom(alice.lastResortAddress, alice.wallet.address, 8, { from: alice.lastResortAddress })
+    // assert.equal(alice.wallet.address, await testerc721.ownerOf(8), 'Transfer of ERC721 token 8 to alice.wallet succesful')
+    // // alice transfers tokens to bob
+    // await tokenTransfer(
+    //   {
+    //     wallet: alice,
+    //     operationType: ONEConstants.OperationType.TRANSFER_TOKEN,
+    //     tokenType: ONEConstants.TokenType.ERC721,
+    //     contractAddress: testerc721.address,
+    //     tokenId: 8,
+    //     dest: bob.wallet.address,
+    //     amount: 1
+    //   }
+    // )
+    // // transfer ERC721 tokens from alice.lastResortAddress (which owns the tokens) to alices wallet
+    // // TODO review the bytes value we are passing in safeTransferFrom (currently using ONEUtil.hexStringToBytes('5') )
+    // await testerc1155.safeTransferFrom(alice.lastResortAddress, alice.wallet.address, 8, 8, ONEUtil.hexStringToBytes('5'), { from: alice.lastResortAddress })
+    // // alice transfers tokens to bob
+    // await tokenTransfer(
+    //   {
+    //     wallet: alice,
+    //     operationType: ONEConstants.OperationType.TRANSFER_TOKEN,
+    //     tokenType: ONEConstants.TokenType.ERC1155,
+    //     contractAddress: testerc1155.address,
+    //     tokenId: 8,
+    //     dest: bob.wallet.address,
+    //     amount: 3
+    //   }
+    // )
+  })
+})
+
+// makeWallet uses an index and unlocked web3.eth.account and creates and funds a ONEwallet
+const makeWallet = async (accountIndex, lastResortAddress) => {
+  // create wallet
+  const EFFECTIVE_TIME = Math.floor(Date.now() / INTERVAL / 6) * INTERVAL * 6 - DURATION / 2
+  const TEN_ETH = unit.toWei('10', 'ether')
```

Comment:
I suggest using a small amount for testing in cases that some tests won't work with snapshot-revert pattern